AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -15,6 +15,7 @@ protocol MeasurementStreamStorage {
     func updateSessionEndtime(_ endTime: Date, for sessionUUID: SessionUUID) throws
     func updateSessionFollowing(_ sessionStatus: SessionFollowing, for sessionUUID: SessionUUID)
     func existingMeasurementStream(_ sessionUUID: SessionUUID, name: String) throws -> MeasurementStreamLocalID?
+    func save() throws
```

Comment:
That's an interesting architectural problem here. It's perfectly fine to have a `save` method on the storage, but it poses a risk of someone forgetting it after making changes. It can be hard to track bugs due to that. I'm wondering if maybe we should be forcing those saves by enclosing other calls in some form of a batch updates block or something. Callsite would look like this:
```Swift
measurementStoarge.performUpdates { updateContext in
    for i in 0..<ONE_BILLION { updateContext.addMeasurement(...) }
}
```

This solution would require MeasurementStreamStorage to be designed a bit differently:

```Swift
protocol MasurementStreamStorage {
    func performUpdates(_ updateBlock: (MeasurementStreamStorageContext) -> Void)
}

protocol MeasurementStreamStorageContext {
    func addMeasurement(_ measurement: Measurement, toStreamWithID id: MeasurementStreamLocalID) throws
}
```

This way we can ensure that `context.save` is called each time we're leaving the `updateBlock` without forcing users of the storage to remember it by themselves.

this is just a thought, it can 100% just stay like this